src/dune/action_exec.ml

```diff
@@ -387,6 +399,40 @@ and exec_list ts ~ectx ~eenv =
     | Need_more_deps _ as need -> Fiber.return need
     | Done -> exec_list rest ~ectx ~eenv )
 
+and exec_pipe outputs ts ~ectx ~eenv =
+  let tmp_file () =
+    Path.of_string
+      (Filename.temp_file "pipe-action-"
+         ("." ^ Action.Outputs.to_string outputs))
+  in
+  let multi_use_eenv =
+    match outputs with
+    | Outputs -> eenv
```

Comment:
That looks wrong

Reply:
Never-mind, it is correct

Reply:
Do you want me to add a comment there to clarify why it's done like this?

Reply:
No that's fine, I just read it too fast.